test-data/unit/semanal-errors.test

```diff
@@ -1451,3 +1451,10 @@ heterogenous_tuple: Tuple[Unpack[Tuple[int, str]]]
 homogenous_tuple: Tuple[Unpack[Tuple[int, ...]]]
```

Comment:
Also add at least one AST dump test case to `test-data/unit/semanal-types.test`.